### PR TITLE
Handle docker tag -f deprecation

### LINKED ISSUE #781: Build failing for docker 1.12 due to `docker tag -f` flag which is now removed
We've done it folks. Deprecation warning since 1.10 and have we done? Ignore it!

Fix should be just to remove `-f` altogether from all references of `docker tag` such as the offending https://github.com/convox/rack/blob/c533520b5e587e6e44253fb0240bab618c355bc4/api/manifest/manifest.go#L349 that prevents `convox start` from working with this error:

```
Successfully built 52b83b62a098
RUNNING: docker tag -f convox-hkrfeetpkt web-dev/web
unknown shorthand flag: 'f' in -f
See 'docker tag --help'.
ERROR: exit status 1
```

### PR DESCRIPTION
**Pull request by awsmsrc**
fixes #781 

## Quick CLI Release Playbook
- [x] Code review
- [ ] Merge into master
- [ ] Release CLI

### CODE REVIEW

File: cmd/convox/start.go [hunk 1]
```diff
@@ -55,6 +57,29 @@ func cmdStart(c *cli.Context) error {
 		return stdcli.ExitError(errors.New("could not connect to docker daemon, is it installed and running?"))
 	}
 
+	if os.Getenv("PROVIDER") != "test" {
```
Comment (ddollar) [inline cmd/convox/start.go]: Let's not do this. It makes CI much less useful.

File: cmd/convox/start.go [hunk 2]
```diff
@@ -55,6 +57,29 @@ func cmdStart(c *cli.Context) error {
...
+			return stdcli.ExitError(err)
+		}
+
+		minDockerVersion, err := docker.NewAPIVersion("1.9")
+		e, err := dockerVersionTest.Version()
+		if err != nil {
+			return stdcli.ExitError(err)
+		}
+
+		currentVersionParts := strings.Split(e.Get("Version"), ".")
```
Comment (ddollar) [inline cmd/convox/start.go]: Can we just feed the version in instead of splitting and recombining?

File: cmd/convox/start.go [hunk 3]
```diff
@@ -55,6 +57,29 @@ func cmdStart(c *cli.Context) error {
...
+		currentVersion, err := docker.NewAPIVersion(fmt.Sprintf("%s.%s", currentVersionParts[0], currentVersionParts[1]))
+		if err != nil {
+			return stdcli.ExitError(err)
+		}
+
+		if !(currentVersion.GreaterThanOrEqualTo(minDockerVersion)) {
+			return stdcli.ExitError(errors.New("You're version of docker is out of date (min: 1.9)"))
```
Comment (ddollar) [inline cmd/convox/start.go]: Your not you're. 

Is 1.9 the version where the tag behavior changed?

File: cmd/convox/start.go (hunk 2 above)
Comment (awsmsrc) [inline cmd/convox/start.go]: we can not without using a different library or writing our own.
My version is coming out as "1.12.0-rc2" which throws an error something like "0-rc2 can not be converted to an integer"

File: cmd/convox/start.go (hunk 3 above)
Comment (awsmsrc) [inline cmd/convox/start.go]: https://docs.docker.com/v1.9/engine/misc/deprecated/ looks like it was actually deprecated in 1.8.

AWS is using 1.9 and there was only 3 months difference between the 1.8 and 1.9 release.

Happy to change to 1.8 if you prefer?

**Comment by mattmanning** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;23eKD3uy&#x2F;268-docker-tag-f"> docker tag -f</a>

**Comment by mattmanning** [comment]
:+1: working well for me on Docker for Mac 1.12.0-rc2-beta16

**Comment by MiguelMoll** [comment]
Working good on Linux for me. :+1:

**Pull request merged**